google/cloud/forseti/services/model/importer/importer.py

```diff
@@ -1091,6 +1145,50 @@ def _convert_organization(self, organization):
         self._add_to_cache(organization, org)
         self.session.add(org)
 
+    def _is_resource_unique(self, type_name):
+        """Check to see if the session contains Resource with
+        primary key = type_name.
+
+        Args:
+            type_name (str): The type name (Primary key of the resource table).
+
+        Returns:
+            bool: Whether or not session contains Resource with
+                primary key = type_name.
+        """
+
+        # one_or_none returns None if the query selects no rows.
+        exists = self.session.query(
```

Comment:
This needs optimization. Doing a query every time might be costly. Can we use the resource_cache to check duplication instead?

Reply:
Updated, do you mind talking another pass at this?